client/containers/Pub/PubHeader/ReviewHeaderSticky.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useState } from 'react';
+import { Button } from '@blueprintjs/core';
+
+import { usePageContext } from 'utils/hooks';
+import { pubUrl } from 'utils/canonicalUrls';
+import { Icon } from 'components';
+import { apiFetch } from 'client/utils/apiFetch';
+import { useLocalStorage } from 'client/utils/useLocalStorage';
+import { getEmptyDoc } from 'client/components/Editor';
+import { DocJson } from 'types';
+
+import { usePubContext } from '../pubHooks';
+import Review from './Review/Review';
+import ReviewerDialog from './Review/ReviewerDialog';
+
+require('./reviewHeaderSticky.scss');
+
+const ReviewHeaderSticky = () => {
+	const { pubData, updatePubData } = usePubContext();
+	const {
+		communityData,
+		scopeData: { activePermissions, memberData },
+		loginData: { fullName },
+	} = usePageContext();
+
+	const [visible, setVisible] = useState(false);
+	const [reviewTitle, setReviewTitle] = useState('Untitled Review');
+	const [reviewerName, setReviewerName] = useState('');
+	const [createError, setCreateError] = useState(undefined);
+	const [isLoading, setIsLoading] = useState(false);
+	const [createdReview, setCreatedReview] = useState(false);
+	const [showReview, setShowReview] = useState(false);
+	const [reviewNumber, setReviewNumber] = useState(0);
+
+	// creates a docjoson object in local store, provides state handlers as well
+	const { value: review, setValue: setReview } = useLocalStorage<DocJson>({
+		initial: () => getEmptyDoc(),
+		communityId: communityData.id,
+		featureName: 'new-review-editor',
+		path: [`pub-${pubData.id}`],
+		debounce: 100,
+	});
+
+	const updatingReviewDoc = (doc: DocJson) => {
+		setReview(doc);
+	};
+
+	const url = new URL(window.location.href);
+	const query = new URLSearchParams(url.search);
+	const handleSubmit = () => {
+		setIsLoading(true);
+		apiFetch
+			.post('/api/reviews', {
+				communityId: communityData.id,
+				pubId: pubData.id,
+				reviewContent: review,
+				title: reviewTitle,
+				accessHash: query.get('access'),
+				reviewerName,
+			})
+			.then((reviewRes) => {
+				updatePubData((currentPubData) => {
+					return currentPubData.reviews
+						? {
+								reviews: [...currentPubData.reviews, reviewRes],
+						  }
+						: {
+								reviews: [reviewRes],
+						  };
+				});
+				setIsLoading(false);
+				setReviewNumber(reviewRes.number);
+				setReview(getEmptyDoc());
+				setCreatedReview(true);
+			})
+			.catch((err) => {
+				setIsLoading(false);
+				setCreateError(err);
+			});
+	};
+
+	const renderReview = () => (
+		<Review
+			communityData={communityData}
+			onSubmit={() => setVisible(true)}
+			isLoading={isLoading}
+			review={review}
+			updateReview={updatingReviewDoc}
```

Comment:
We could remove the intermediate `updatingReviewDoc ` function and just pass `setReview`

Reply:
yeah, i needed that for the saving UI mentioned by Deepak in the [designs](https://www.figma.com/file/vkVeMt7JF2HdzY86RENIkX/Review-Designs?node-id=13%3A2) but im not about where it is now 
![image](https://user-images.githubusercontent.com/34730449/187726466-0b463b90-ede3-4d35-907c-2bb831372553.png)